File: Builder/button/ButtonBuilder.java
package button;

public abstract class ButtonBuilder {
    protected int height;
    protected int width;
    protected String text;
    protected Color backgroundColor;
    protected Color textColor;
    protected String action;
    public ButtonBuilder setHeight(final int height) {
        this.height = height;
        return this;
    }

    public ButtonBuilder setWidth(final int width) {
        this.width = width;
        return this;
    }
    public ButtonBuilder setText(final String text) {
        this.text = text;
        return this;
    }
    public ButtonBuilder setBackgroundColor(final Color backgroundColor) {
        this.backgroundColor = backgroundColor;
        return this;
    }
    public ButtonBuilder setTextColor(final Color textColor) {
        this.backgroundColor = backgroundColor;
        return this;
    }
    public ButtonBuilder setAction(final String action) {
        this.action = action;
        return this;
    }
    public abstract Button build();
}

File: BuilderKotlin/button/ButtonBuilder.kt
package button

abstract class ButtonBuilder protected constructor(
    var height: Int? = null,
    var width: Int? = null,
    var text: String? = null,
    var action: String? = null,
    var backgroundColor: Color? = null,
    var textColor: Color? = null
) {

    fun setHeight(height: Int) = apply { this.height = height }
    fun setWidth(width: Int) = apply { this.width = width }
    fun setText(text: String) = apply { this.text = text }
    fun setAction(action: String) = apply { this.action = action }
    fun setBackgroundColor(backgroundColor: Color) = apply { this.backgroundColor = backgroundColor }
    fun setTextColor(textColor: Color) = apply { this.textColor = textColor }

    abstract fun build() : Button
}